multi_gpu/four/model_parallel.py

```diff
@@ -0,0 +1,162 @@
+####################
+"""Model Adapted from https://github.com/jamespengcheng/PyTorch-CNN-on-CIFAR10/blob/master/ConvNetClassifier.py"""
+####################
+
+
+import torch
+import torchvision
+import torchvision.datasets as datasets
+import torchvision.transforms as transforms
+from torch.autograd import Variable
+import torch.nn as nn
+import torch.nn.functional as F
+
+#import wandb
+
+#wandb.init(project="FOUR")
+
+# Configuration area
+#############################
+epochs = 10
+batch_size = 256
+learning_rate = 0.001
+num_workers = 4
+#############################
+use_cuda = True
+device = torch.cuda.device("cuda" if use_cuda else "cpu")
+
+transform_train = transforms.Compose([
+    transforms.RandomCrop(32, padding=4),
+    transforms.RandomHorizontalFlip(),
+    transforms.ToTensor(),
+    transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
+])
+
+transform_test = transforms.Compose([
+    transforms.ToTensor(),
+    transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
+])
+
+cifar_trainset = datasets.CIFAR10(root='./data', train=True, download=True, transform=transform_train)
+trainset_loader = torch.utils.data.DataLoader(cifar_trainset, batch_size=batch_size,shuffle=True, num_workers=num_workers)
+
+cifar_testset = datasets.CIFAR10(root='./data', train=False, download=True, transform=transform_test)
+testset_loader = torch.utils.data.DataLoader(cifar_testset, batch_size=batch_size, shuffle=False, num_workers=num_workers)
+
+class ConvNet(nn.Module):
+    def __init__(self):
+        super(ConvNet, self).__init__()
+        self.conv1 = nn.Conv2d(in_channels=3, out_channels=48, kernel_size=(3,3), padding=(1,1))
+        self.conv1 = nn.DataParallel(self.conv1)
+        self.conv2 = nn.Conv2d(in_channels=48, out_channels=96, kernel_size=(3,3), padding=(1,1))
+        self.conv2 = nn.DataParallel(self.conv2)
+
+        self.conv3 = nn.Conv2d(in_channels=96, out_channels=192, kernel_size=(3,3), padding=(1,1))
+        self.conv3 = nn.DataParallel(self.conv3)
+
+        self.conv4 = nn.Conv2d(in_channels=192, out_channels=256, kernel_size=(3,3), padding=(1,1))
+        self.conv4 = nn.DataParallel(self.conv4)
+        
+        self.conv5 = nn.Conv2d(in_channels=256, out_channels=512, kernel_size=(3,3), padding=(1,1))
+        self.conv5 = nn.DataParallel(self.conv5)        
```

Comment:
You don't need to wrap individual layer with DataParallel. You can wrap the whole model and it will cover all the layers. See comment below.